task/common/machine/script.go

```diff
@@ -7,25 +7,32 @@ import (
 	"strings"
 	"time"
 
+	"github.com/alessio/shellescape"
+
 	"terraform-provider-iterative/task/common"
 )
 
-func Script(script string, variables common.Variables, timeout time.Duration) string {
-	var environment string
+func Script(script string, credentials *map[string]string, variables common.Variables, timeout time.Duration) string {
+	timeoutString := strconv.Itoa(int(timeout / time.Second))
+	if timeout <= 0 {
+		timeoutString = "infinity"
+	}
+
+	environment := ""
 	for name, value := range variables.Enrich() {
 		escaped := strings.ReplaceAll(value, `"`, `\"`) // FIXME: \" edge cases.
 		environment += fmt.Sprintf("%s=\"%s\"\n", name, escaped)
 	}
 
-	timeoutString := strconv.Itoa(int(timeout / time.Second))
-	if timeout <= 0 {
-		timeoutString = "infinity"
+	exportCredentials := ""
+	for name, value := range *credentials {
+		exportCredentials += "export " + shellescape.Quote(name+"="+value) + "\n"
 	}
 
 	return fmt.Sprintf(
 		`#!/bin/bash
-sudo mkdir --parents /tmp/tpi-task
-chmod u=rwx,g=rwx,o=rwx /tmp/tpi-task
+sudo mkdir --parents /opt/task/directory
+chmod u=rwx,g=rwx,o=rwx /opt/task/directory
```

Comment:
any reason for:

- `tpi-task` -> `task`
- not using `/var/cache/`?

Reply:
* Simplicity: it's unlikely we find any `task` directory in any of these locations. 🤷🏼 
* Finally I thought that, if we were going to store environment variables and other files in the future, `/var/cache` wouldn't be the right choice either. 🤔